### PR TITLE
Revert "[revert] test with Xarray PR branch"

### PR DESCRIPTION
**Pull request by dcherian**
This reverts commit 4a59ffa2cdfcb2d4fc662f451d153d68cf77a5b2.

### CODE REVIEW

File: ci/no-dask.yml [hunk 1]
```diff
@@ -21,5 +21,4 @@ dependencies:
   - toolz
   - numba
   - numbagg>=0.3
-  - pip:
-      - git+https://github.com/dcherian/xarray.git@flox-preserve-dtype
+  - xarray
```
Comment (keewis) [inline ci/no-dask.yml:24]: might need to do the same for `env-numpy1`?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -102,7 +102,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          repository: "dcherian/xarray"
+          repository: "pydata/xarray"
```
Comment (keewis) [inline .github/workflows/ci.yaml:105]: is there a reason why you clone `xarray` `main`, then install `flox` from a github link?

I'd have expected to clone `flox`, then install `xarray` from github (or from the nightlies), and then finally install `flox` from local.

Comment (dcherian) [reply .github/workflows/ci.yaml:105]: This job runs the *xarray* tests, that's why I clone it.

**Comment by dcherian** [comment]
Right, we still need an xarray release for this to work as-is

**Pull request merged**